bidict/_bidict.py

```diff
@@ -7,58 +7,75 @@ class bidict(BidirectionalMapping, MutableMapping):
     """
     Mutable bidirectional map type.
     """
+
     def _del(self, key):
         val = self._fwd[key]
         del self._fwd[key]
         del self._bwd[val]
 
     def __delitem__(self, key):
+        """
+        Analogous to dict.__delitem__(), keeping bidirectionality intact.
+        """
         self._del(key)
 
     def __setitem__(self, key, val):
-        self._put(key, val)
+        """
+        Inserts the given mapping iff it wouldn't overwrite an existing key
```

Comment:
I'm not sure about this language of "overwrite an existing key". To me this is more confusing. Perhaps "Set the value for key to val, replacing any previous key that maps to this value"?

Also, PEP257.